### ISSUE TITLE
Count over column projection after GroupBy yields wrong results

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#26938]
In 6.0, we generate `COUNT(*)` for column projection after GroupBy, whereas in 5.0 we generated COUNT with the column name:

```c#
_ = ctx.Set<Customer>()
    .GroupBy(o => o.City)
    .Select(g => new { City = g.Key, Faxes = g.Select(g2 => g2.Fax).Count() })
    .ToList();
```

6.0:

```sql
SELECT [c].[City], COUNT(*) AS [Faxes]
FROM [Customers] AS [c]
GROUP BY [c].[City]
```

5.0:

```sql
SELECT [c].[City], COUNT([c].[Fax]) AS [Faxes]
FROM [Customers] AS [c]
GROUP BY [c].[City]
```

In all databases (SQL Server, SQLite, PostgreSQL, MySQL), `COUNT(x)` counts only non-null x, whereas `COUNT(*)` counts total rows. Unless I'm mistaken, this means that 5.0 returned wrong results, and 6.0 fixed this.

/cc @smitpatel

Originally raised in https://github.com/npgsql/efcore.pg/issues/2143 by @sa13m.

**Comment by roji** [comment]
@smitpatel tried to look at some non-GroupBy scenarios, I'm not seeing COUNT(column):

```c#
_ = ctx.Blogs
    .GroupBy(b => b.Id).Select(g => g.Select(g => g.Name).Count())
    .ToList();

_ = ctx.Blogs.Select(b => b.Name).Count();
```

EF Core 6.0:

```sql
SELECT COUNT(*)
FROM [Blogs] AS [b]
GROUP BY [b].[Id]

SELECT COUNT(*)
FROM [Blogs] AS [b]
```

EF Core 5.0:

```sql
SELECT COUNT([b].[Name])
FROM [Blogs] AS [b]
GROUP BY [b].[Id]

SELECT COUNT(*)
FROM [Blogs] AS [b]
```

Any specific scenario you have in mind where this translates differently?

**Comment by roji** [comment]
Design decision: the new 6.0 behavior is the one we want, so this is "fixed in 6.0" (regression tests added in #26939 for 7.0)
